core/jvm/src/main/scala/fs2/compress2.scala

```diff
@@ -0,0 +1,862 @@
+      val bytes = string.replaceAll("\u0000", " ").getBytes(StandardCharsets.ISO_8859_1)
+      crc32.update(bytes)
+      crc32.update(zeroByte)
+      bytes
+    }
+    val crc32Value = crc32.getValue
+    val crc16 = Array[Byte](
+      (crc32Value & 0xff).toByte,
+      ((crc32Value >> 8) & 0xff).toByte
+    )
+    Stream.chunk(asChunkBytes(header)) ++
+      fileNameEncoded
+        .map(bytes => Stream.chunk(asChunkBytes(bytes)) ++ Stream.emit(zeroByte))
+        .getOrElse(Stream.empty) ++
+      commentEncoded
+        .map(bytes => Stream.chunk(asChunkBytes(bytes)) ++ Stream.emit(zeroByte))
+        .getOrElse(Stream.empty) ++
+      Stream.chunk(asChunkBytes(crc16))
+  }
+
+  private def _gzip_trailer[F[_]](deflater: Deflater, crc32: CRC32): Stream[F, Byte] = {
+    // See RFC 1952: https://www.ietf.org/rfc/rfc1952.txt
+    val crc32Value = crc32.getValue
+    val bytesIn = deflater.getTotalIn
+    val trailer = Array[Byte](
+      (crc32Value & 0xff).toByte, // CRC-32: Cyclic Redundancy Check
+      ((crc32Value >> 8) & 0xff).toByte,
+      ((crc32Value >> 16) & 0xff).toByte,
+      ((crc32Value >> 24) & 0xff).toByte,
+      (bytesIn & 0xff).toByte, // ISIZE: Input size
+      ((bytesIn >> 8) & 0xff).toByte,
+      ((bytesIn >> 16) & 0xff).toByte,
+      ((bytesIn >> 24) & 0xff).toByte
+    )
+    Stream.chunk(asChunkBytes(trailer))
+  }
+
+  final case class GunzipResult[F[_]](
+      content: Stream[F, Byte],
+      modificationTime: Option[Instant] = None,
+      fileName: Option[String] = None,
+      comment: Option[String] = None
+  )
+
+  /**
+    * Returns a pipe that incrementally decompresses input according to the GZIP
+    * format as defined by RFC 1952 at https://www.ietf.org/rfc/rfc1952.txt. Any
+    * errors in decompression will be sequenced as exceptions into the output
+    * stream. Decompression is handled in a streaming and async fashion without
+    * any thread blockage.
+    *
+    * The chunk size here is actually really important. Matching the input stream
+    * largest chunk size, or roughly 8 KB (whichever is larger) is a good rule of
+    * thumb.
+    *
+    * @param bufferSize The bounding size of the input buffer. This should roughly
+    *                   match the size of the largest chunk in the input stream.
+    *                   This will also be the chunk size in the output stream.
+    *                    Default size is 32 KB.
+    * @return    Returns a stream containing a single GzipResult containing the
+    *            properties of the content stream. Continue to stream the content
+    *            by flat mapping it:
+    *            stream
+    *             .through(gunzip[IO]())
+    *             .flatMap { gunzipResult =>
+    *               // Access properties here.
+    *               gunzipResult.content
+    *             }
+    */
+  def gunzip[F[_]](
+      bufferSize: Int = 1024 * 32
+  )(
+      implicit SyncF: Sync[F],
+      RaiseThrowableF: RaiseThrowable[F]
```

Comment:
Note, we don't need `RaiseThrowable[F]` here since we have `Sync[F]`.

Reply:
Agreed. Thanks for highlighting this. The change has been applied, the branch rebased and the pull request green.